js/packages/teams-ai/src/planners/ActionPlanner.ts

```diff
@@ -0,0 +1,214 @@
+/**
+ * @module teams-ai
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Planner, Plan } from './Planner';
+import { TurnState } from '../TurnState';
+import { TurnContext } from 'botbuilder';
+import { AI } from '../AI';
+import { PromptTemplate, PromptManager } from '../prompts';
+import { PromptCompletionModel, PromptResponse } from '../models';
+import { PromptResponseValidator } from '../validators';
+import { Memory } from '../MemoryFork';
+import { LLMClient } from './LLMClient';
+import { Tokenizer } from '../tokenizers';
+import { Utilities } from '../Utilities';
+import { DefaultAugmentation } from '../augmentations';
+
+export interface ActionPlannerOptions<TState extends TurnState = TurnState> {
+    model: PromptCompletionModel;
+    prompts: PromptManager;
+    defaultPrompt: string|((context: TurnContext, state: TState, planner: ActionPlanner<TState>) => Promise<PromptTemplate>);
```

Comment:
Can we leave documentation on the possible string defaultPrompt values? monologue, chat, etc.